packages/serverpod_client/lib/src/serverpod_client_shared.dart

```diff
@@ -22,6 +22,25 @@ enum StreamingConnectionStatus {
   waitingToRetry,
 }
 
+/// Context for a method call.
+class MethodCallContext {
+  /// Name of the called endpoint.
+  final String endpointName;
+
+  /// Name of the called endpoint method.
+  final String methodName;
+
+  /// Arguments passed to the method.
+  final Map<String, dynamic> arguments;
+
+  /// Creates a new [MethodCallContext].
+  MethodCallContext({
```

Comment:
This could have been `const`, but it may not have a practical implication here as we are never creating this with const values.

Reply:
Aah good call! Thanks for the tip! 👍